sdk/python/feast/offline_store.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List, Optional, Tuple
+
+import pyarrow
+
+from feast.feature_view import FeatureView
+
+
+class OfflineStore(ABC):
+    """
+    OfflineStore is a non-user-facing object used for all interaction between Feast and the service used for offline storage of features. Currently BigQuery is supported.
```

Comment:
Nit: I think we can simplify this. Non-user-facing isnt really relevant.

What about `An OfflineStore is used for the storage and retrieval of historical feature data`

Reply:
I think non-user-facing is important since these docstrings get turned into our API documentation - we don't want users to use this class to trigger ingestion or historical retrieval

Reply:
But wouldn't that apply to the majority of our code?

Reply:
I'm not sure why we have API documentation for our internal classes tbh

Reply:
In this specific case we expect actual developers to work with these classes. Documenting it is a very small amount of effort for us and makes it a lot more pleasant to navigate and read a code base.